ash/src/vk.rs

```diff
@@ -49,4 +49,15 @@ pub trait Handle {
     const TYPE: ObjectType;
     fn as_raw(self) -> u64;
     fn from_raw(_: u64) -> Self;
+
+    /// Returns whether the handle is a `NULL` value.
+    ///
+    /// # Example
+    ///
+    /// ```
+    /// # use ash::vk::{Handle, Instance};
+    /// let instance = Instance::null();
+    /// assert!(instance.is_null());
+    /// ```
+    fn is_null(self) -> bool;
```

Comment:
Maybe have a provided impl of `self.as_raw() == 0`?

Reply:
Thought about that as well, but ended up not doing that. Wouldn't hurt to have it though, I guess?

Reply:
Oh apparently this is only allowed if we add `Sized` to the trait bound of `Handle`:

```rust
error[E0277]: the size for values of type `Self` cannot be known at compilation time
  --> ash/src/vk.rs:62:16
   |
62 |     fn is_null(self) -> bool {
   |                ^^^^ doesn't have a size known at compile-time
   |
help: consider further restricting `Self`
   |
62 |     fn is_null(self) -> bool where Self: std::marker::Sized {
   |                              ++++++++++++++++++++++++++++++
help: function arguments must have a statically known size, borrowed types always have a known size
   |
62 |     fn is_null(&self) -> bool {
   |                +
```

Which is fine.

Reply:
The main benefit of a provided impl is that now we can drop the explicit impls from the macro.

Reply:
D'oh, I forgot to save.